src/giskardpy/casadi_wrapper.py

```diff
@@ -2147,3 +2147,11 @@ def substitute(expression, old_symbols, new_symbols):
     new_symbols = Expression([_to_sx(s) for s in new_symbols]).s
     sx = ca.substitute(sx, old_symbols, new_symbols)
     return type(expression)(sx)
+
+
+def matrix_inverse(a):
```

Comment:
the implementation doesn't match the type hint. you should check if its a transmatrix and then call its inverse method and if it's not, you should turn the result into an Expression